### PR TITLE
Remove Gulp as a task runner

### LINKED ISSUE #229: Move away from Gulp
The nagging NPM security alerts for the repo all come from our dependency on Gulp for building. As much as I love it compared to the alternatives, we should consider moving away from it in order not to cloud more meaningful alerts, should they occur.

### PR DESCRIPTION
**Pull request by philliphoff**
Removes Gulp as a task runner in favor of npm scripts.  As Gulp is essentially "done", it's not receiving updates which means it's a continual source of corporate governance alerts, irrelevant as they may be.

> NOTE: As the localization pipeline of `vscode-nls-dev` is Gulp-based, we lose that ability for now, but technically we weren't really using it anyway.  (The source is still prepped for localization, but we just won't do the transformation/inclusion of the localization files, which we don't have anyway.) We can revisit this if and when we get to 1.0 (general availability).

Resolves #229.

@bwateratmsft I hope your happy now.  :-)

### CODE REVIEW

**Comment by bwateratmsft** [comment]
> @bwateratmsft I hope your happy now. :-)

😈

File: gulpfile.ts [hunk 1]
```diff
@@ -110,14 +110,6 @@ function addI18nTask() {
         .pipe(gulp.dest('.'));
```
Comment (bwateratmsft) [inline gulpfile.ts:110]: Any reason to keep this file anymore?

**Review by bwateratmsft — APPROVED** [review]

File: gulpfile.ts (hunk 1 above)
Comment (philliphoff) [reply gulpfile.ts]: My subconcious just wouldn't let me let it go.  Removed.  <sigh> :-)

Comment (bwateratmsft) [reply gulpfile.ts]: image

_Phil letting Gulp go_

Comment (philliphoff) [reply gulpfile.ts]: Gulp will make a comeback.  I just *know* it!

**Pull request merged**